drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/data_warehouse_upload_job.rb

```diff
@@ -12,18 +12,32 @@ class DataWarehouseUploadJob < BaseJob
 
     attr_accessor :state
 
-    def perform(mode = 'clients_with_mci_ids_and_address')
+    def perform(mode)
```

Comment:
maybe `mode` should be an array of method names?  (or I guess `daily_uploads` could just be a method that calls the whole set of known methods. 
```
def perform(methods)
   methods.each do |method|
     next unless known?(method) # maybe raise?
    
     call(method)
end

def known?(method)
  known_methods.include?(method)
end

def known_methods
  [
    'clients_with_mci_ids_and_address',
    ...
  ].freeze
end
```

Reply:
Yeah, definitely a chance for misalignment. I was trying to get away from the fact that the methods themselves need to be listed in `export.rake` which means that when you add a new method you need to add it to both files, which was always annoying (and also has a potential for misalignment).

I think with your suggestion, we would still need to list out each upload in the rake task? Or is there something I'm missing?

Reply:
Ah I see what you mean, thanks. I think I addressed this adequately with https://github.com/greenriver/hmis-warehouse/pull/4602/commits/524cbc6dcec0dc8de81ee6a7ac6a8e0834e83f1f